metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorDatabaseService.java

```diff
@@ -159,17 +160,12 @@ public List<QualifiedName> listNames(
         @Nullable final Pageable pageable
     ) {
         try {
-            List<QualifiedName> qualifiedNames = polarisStoreService.getAllDatabases().stream()
-                .map(d -> QualifiedName.ofDatabase(name.getCatalogName(), d.getDbName()))
+            final String dbPrefix = prefix == null ? "" : prefix.getDatabaseName();
+            final List<QualifiedName> qualifiedNames = polarisStoreService.getDatabaseNames(
+                dbPrefix, sort, 1000)
```

Comment:
Should we try a larger page size like 5000 or 10000? That's what we bumped it to for table names and it seems it could be more efficient.